app/src/main/java/emu/skyline/MainActivity.kt

```diff
@@ -217,15 +217,42 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun getDataItems() = mutableListOf<DataItem>().apply {
-        appEntries?.let { entries ->
-            val formats = formatFilter?.let { listOf(it) } ?: formatOrder
-            for (format in formats) {
-                entries[format]?.let {
-                    add(HeaderItem(format.name))
-                    it.forEach { entry -> add(AppItem(entry)) }
+        if (preferenceSettings.groupByFormat) {
+            appEntries?.let { entries ->
+                val formats = formatFilter?.let { listOf(it) } ?: formatOrder
+                for (format in formats) {
+                    entries[format]?.let {
+                        add(HeaderItem(format.name))
+                        for (entry in sortGameList(it)) {
+                            add(AppItem(entry))
+                        }
+                    }
+                }
+            }
+        } else {
+            val gameList = mutableListOf<AppEntry>()
+            appEntries?.let { entries ->
+                val formats = formatFilter?.let { listOf(it) } ?: formatOrder
+                for (format in formats) {
+                    entries[format]?.let {
+                        it.forEach { entry -> gameList.add(entry) }
+                    }
                 }
             }
+            for (entry in sortGameList(gameList.toList())) {
+                add(AppItem(entry))
+            }
+        }
```

Comment:
```suggestion
        appEntries?.let { entries ->
            val formats = formatFilter?.let { listOf(it) } ?: formatOrder

            if (preferenceSettings.groupByFormat) {
                for (format in formats) {
                    entries[format]?.let {
                        add(HeaderItem(format.name))
                        for (entry in sortGameList(it)) {
                            add(AppItem(entry))
                        }
                    }
                }
            } else {
                for (format in formats) {
                    entries[format]?.let {
                        it.forEach { entry -> gameList.add(entry) }
                    }
                }
                sortGameList(this)
            }
        }
```
Tried simplifying logic a bit, still a mess because entries are saved in a `Map` per format. I didn't test it though, so make sure to do that.